app/models/invenio_rdm_record_converter.rb

```diff
@@ -286,4 +290,75 @@ def version(content)
 
     "v#{version_number}.0.0"
   end
+
+  def format_publication_date(publication_date)
+    normalize_date(publication_date)
+  end
+
+  def normalize_date(date_string)
+    split_date = date_string.split(/[-,\/ ]/).map(&:downcase)
+    month_names = (split_date & MONTHNAMES)
+    abbr_month_names = (split_date & ABBR_MONTHNAMES)
+
+    # blank and unddated
```

Comment:
each of these branches could potentially be it's own function, but I decided to just use these comments while I worked through it. Originally my intention was to go back and refactor, but I really don't think that's worth the time given how long this took me to complete.